vxquery-cli/src/main/java/org/apache/vxquery/cli/VXQuery.java

```diff
@@ -120,200 +94,236 @@ public static void main(String[] args) throws Exception {
-        Date end = opts.timing ? new Date() : null;
-        if (opts.timing) {
-            timingMessage("Compile time: " + (end.getTime() - start.getTime()) + " ms");
+
+        if (request.isShowTranslatedExpressionTree()) {
+            printField("Translated Expression Tree", response.getTranslatedExpressionTree());
         }
-        if (opts.compileOnly) {
-            return;
+
+        if (request.isShowOptimizedExpressionTree()) {
+            printField("Optimized Expression Tree", response.getOptimizedExpressionTree());
         }
 
-        Module module = compiler.getModule();
-        JobSpecification js = module.getHyracksJobSpecification();
-
-        DynamicContext dCtx = new DynamicContextImpl(module.getModuleContext());
-        js.setGlobalJobDataFactory(new VXQueryGlobalDataFactory(dCtx.createFactory()));
-
-        // Repeat execution for number of times provided in -repeatexec argument
-        for (int i = 0; i < repeatedExecution; ++i) {
-            start = opts.timing ? new Date() : null;
-            runJob(js, writer);
-            // if -timing argument passed, show the starting and ending times
-            if (opts.timing) {
-                end = new Date();
-                long currentRun = end.getTime() - start.getTime();
-                if ((i + 1) > opts.timingIgnoreQueries) {
-                    sumTiming += currentRun;
-                    sumSquaredTiming += currentRun * currentRun;
-                    if (currentRun < minTiming) {
-                        minTiming = currentRun;
-                    }
-                    if (maxTiming < currentRun) {
-                        maxTiming = currentRun;
-                    }
-                }
-                timingMessage("Job (" + (i + 1) + ") execution time: " + currentRun + " ms");
-            }
+        if (request.isShowRuntimePlan()) {
+            printField("Runtime Plan", response.getRuntimePlan());
+        }
+
+//        System.out.println(String.format("Reading results for '%s', result ID: %d", xqFile, response.getResultId()));
+        QueryResultRequest resultRequest = new QueryResultRequest(response.getResultId(), response.getRequestId());
+        resultRequest.setShowMetrics(opts.timing);
+        sendQueryResultRequest(xqFile, resultRequest, this);
+    }
+
+    private void onQueryResultFetchSuccess(String xqFile, QueryResultRequest request, QueryResultResponse response) {
+        if (request.isShowMetrics()) {
+            String metrics = String.format("Elapsed Time:\t%d", response.getMetrics().getElapsedTime());
+            printField("Result Reading Metrics", metrics);
+        }
+
+        printField("Results", response.getResults());
+    }
+
+    private void onQueryFailure(String xqFile, ErrorResponse response) {
+        if (response == null) {
+            System.err.println(String.format("Unable to execute query in %s", xqFile));
+            return;
         }
+
+        System.err.println();
```

Comment:
Seems like we are adding a lot of output to the CLI. Do you think this helps the user?